overlord/stateengine.go

```diff
@@ -68,15 +73,41 @@ func (se *StateEngine) State() *state.State {
 // must not perform long running activities during that operation, though.
 // These should be performed in properly tracked changes and tasks.
 func (se *StateEngine) Ensure() error {
+	if !se.inited {
+		for _, m := range se.managers {
```

Comment:
This will not work if we call AddManager() later. I don't think it's bad but we should not allow AddManager() when se.inited is true